src/index.ts

```diff
@@ -8,11 +8,116 @@ import {
   WalletClient,
 } from "viem";
 import { getParagraphAPI } from "./generated/api";
+import type {
+  GetCoin200,
+  GetPostById200,
+  GetPostsFeed200ItemsItem,
+  GetPostsParams,
+  GetPostsFeedParams,
+  GetPublicationById200,
+  GetUser200,
+} from "./generated/models";
+import { setCurrentApiKey } from "./mutator/custom-axios";
 import { base } from "viem/chains";
 import { ADDRESSES } from "@whetstone-research/doppler-sdk";
 import { signPermit } from "doppler-router/dist/Permit2";
 import { CommandBuilder } from "doppler-router";
 
+/**
+ * Configuration options for the Paragraph API client.
+ */
+export interface ParagraphAPIOptions {
```

Comment:
this is a massive file -- I'd suggest separating these interfaces/types into a new types.ts, then separating each resource into it's own file (publications.ts, etc); this index would import then re-export